sceptre/cli/diff.py

```diff
@@ -122,7 +124,7 @@ def diff_command(
     with execution_context:
         diffs: Dict[Stack, StackDiff] = plan.diff(stack_differ)
 
-    num_stacks_with_diff = output_diffs(diffs.values(), writer_class, sys.stdout, output_format)
+    num_stacks_with_diff = output_diffs(diffs.values(), writer_class, no_colour, sys.stdout, output_format)
```

Comment:
Rather than passing an optional flag like you're doing, I think it would be very easy to have a "ColoredDifflibWriter" class that subclasses DifflibWriter and adds the colors. That way, in line 119ish above, you can select which writer makes more sense and then the everything else would work the same as it always has. Thus you'd be using a polymorphic approach that would be simpler all around and require fewer conditionals. It would also prevent you from having to make changes to the base class that are only relevant to a subclass (which is always a red flag).